src/Crypto/Lol/Applications/SymmSHE.hs

```diff
@@ -276,7 +276,7 @@ switch hint c = rescaleLinearMSD $ untag $ knapsack <$>
 
 -- | Constraint synonym for key switching.
 type KeySwitchCtx gad t m' zp zq zq' =
-  (ToSDCtx t m' zp zq,
+  (ToSDCtx t m' zp zq, NFData (Cyc t m' zq'),
```

Comment:
There shouldn't be any constraints on the entire (Cyc t m' zq'), only on the individual params.